src/components/JSONViewer/index.tsx

```diff
@@ -7,22 +7,20 @@ import {
     returnOneElement,
 } from "./formattingUtils";
 import "./style.css";
+import { useRecipes, useSelectedRecipeId } from "../../state/store";
+import { buildCurrentRecipeObject } from "../../state/utils";
 
-interface JSONViewerProps {
-    title: string;
-    content: string;
-    isEditable: boolean;
-    onChange: (value: string) => void;
-}
+const JSONViewer = (): JSX.Element | null => {
+    const selectedRecipeId = useSelectedRecipeId();
+    const recipes = useRecipes();
 
-const JSONViewer = (props: JSONViewerProps): JSX.Element | null => {
-    const { content } = props;
+    const currentRecipe = recipes[selectedRecipeId];
```

Comment:
this should be a selector in state the returns the recipe data. This component should just get the recipe obj, not have to make any calls to state.